CopilotKit/packages/react-textarea/src/lib/debouncer.ts

```diff
@@ -1,7 +1,7 @@
 export type AsyncFunction<T extends any[]> = (...args: [...T, AbortSignal]) => Promise<void>;
 
 export class Debouncer<T extends any[]> {
-  private timeoutId?: number;
+  private timeoutId?: NodeJS.Timeout;
```

Comment:
this is really strange.

react-textarea targets frontends, so ` NodeJS.Timeout` should throw a build error, and `number` should not (it doesn't on all our machines 🤔)

need to figure out why this error occured

Reply:
@ataibarkai I see very strange. i don't see anything odd with my setup so far

could we replace with `private timeoutId?: ReturnType<typeof setTimeout>;` which will correctly represent the return type of setTimeout in both browser and Node.js environments?

Reply:
@ataibarkai I found that one other contributer run into this issue before: https://github.com/CopilotKit/CopilotKit/issues/64#issuecomment-1820691561

Reply:
Very strange. I like your suggested change as a workaround in the meantime- let’s go with that.

I’ll crate a separate ticket to track the underlying issue

Reply:
Pushed